### PR TITLE
Enable IPv6 by escaping [ and ] caracters to not be encoded

### PR DESCRIPTION
**Pull request by petregeorgee**
This pull request resolves the following issue: 
https://github.com/vaadin/framework/issues/11458

Regards,
George

### CODE REVIEW

**Comment by petregeorgee** [comment]
Hello [jfarcand](/jfarcand)

Can you please help me to resolve this little bug? I am very blocked. 
[https://github.com/vaadin/framework/issues/11458](url)

Thank you so much,
George

**Comment by jfarcand** [comment]
@petregeorgee Can you rebase your fix?

**Comment by petregeorgee** [comment]
@jfarcand rebase done. I don't know why, after the rebase, GitHub told me that those two branches had conflicts and I had to resolve them. 

Thank you,
George

File: modules/javascript/src/main/webapp/javascript/atmosphere.js [hunk 1]
```diff
@@ -3104,12 +3104,17 @@
...
             if(ua.indexOf('MSIE ')>0 || ua.indexOf('Trident/') > 0 || ua.indexOf('Edge/') > 0){
-                return encodeURI(decodeURI(div.firstChild.href));
+                return atmosphere.util.fixedEncodeURI(decodeURI(div.firstChild.href));
             }
             return div.firstChild.href;
         },
+		
+        fixedEncodeURI: function (str) {
+			return encodeURI(str).replace(/%5B/g, '[').replace(/%5D/g, ']');
+		},
```
Comment (jfarcand) [inline modules/javascript/src/main/webapp/javascript/atmosphere.js]: Minor, but can you make sure tabs are properly set (format)

Comment (jfarcand) [reply modules/javascript/src/main/webapp/javascript/atmosphere.js]: Still not workking (the format).

Comment (petregeorgee) [reply modules/javascript/src/main/webapp/javascript/atmosphere.js]: Yes, I know. sorry for these problems.  I had some windows problems. On my computer seems to be ok, but here in github  no...:(. 

Now its ok. 

Thank you so much for your patient

**Comment by petregeorgee** [comment]
Thanks!!

**Comment by petregeorgee** [comment]
Hello @jfarcand.

I only have a little question: in which release these changes will be available? 

Thank you in advance,
George

**Comment by jfarcand** [comment]
2.3.9 should be available on Maven Central as of now.

**Comment by petregeorgee** [comment]
I can't see it.  The latest version of the project is 2.3.8, released on Oct 2018 [link](https://mvnrepository.com/artifact/org.atmosphere.client/javascript) 

Other than that I need to use this dependency: "com.vaadin.external.atmosphere:atmosphere-runtime" for my project. Do you know a person with I can talk about this?

**Comment by jfarcand** [comment]
https://search.maven.org/search?q=atmosphere-javascript

**Pull request merged**